Do not alert lack of maintenance on inactive db

ioguix opened the pull request:
closes #129

ioguix commented:
Hi,

The only question on this PR is what to do with the perfdata when the check returns OK despite the last maintenance is older than the given threshold. From the graphing UI, this could lead to metrics entering the read area without alerts.

Ideas?

Note that this could be solved later to avoid postponing last release infinitely.

Krysztophe commented:
Sounds good to me, except the error messages when the status file is not initialized.

I've tested with `-w 1m` and the service does not raise  WARNING as long as nothing is written.
A simple INSERT,UPDATE,CREATE EXTENSION... raises the WARNING.

Krysztophe commented:
Logically, the durations can attain extremely high values (months?).  I don't see it as a problem. 

Users may be astonished to see a WARNING suddenly raised with a very value, just after some administrative command or a very small change, asking why it wasn't raised before. There should be an hint in the doc.

ioguix commented on check_pgactivity:
```diff
@@ -4701,10 +4723,10 @@ LOOP_DB: foreach my $db (@all_db) {
+        if ( defined $counts{$db} ) {
+            if ($hosts[0]->{'version_num'} >= $PG_VERSION_91 ) {
                 my $delta      = $rs->[1] - $counts{$db}[0];
                 my $delta_auto = $rs->[2] - $counts{$db}[1];
```
fixed! Thanks for the review!

ioguix commented:
Hi @Krysztophe,

Thank you for your review.

I fixed the error you found, added some code comments and user documentation.

Let me know if it's OK for you.

Regards,

Krysztophe commented:
Tested again on 8.2, 8.4, 9.1, this seems ok.
As said somewhere else: in 8.2, pg_stat_user_tables depends on `stats_*` parameters that are off by default, so I'm afraid that the service could decide wrongly that nothing happens in the DB if it just looks at this view.

ioguix commented:
> in 8.2, pg_stat_user_tables depends on stats_* parameters that are off by default

This is a good catch. We should probably focus more often on services compatibility with current GUC values...Not just for very old release.

I introduced `is_guc()` to check this. Hopefully it will be used frequently in future releases.

Krysztophe commented:
Looks good to me!
